src/spikeinterface/core/sortingresult.py

```diff
@@ -0,0 +1,1324 @@
+        save: bool, default True
+            It the extension can be saved then it is saved.
+            If not then the extension will only live in memory as long as the object is deleted.
+            save=False is convinient to try some parameters without changing an already saved extension.
+
+        **params: 
+            All other kwargs are transimited to extension.set_params()
+
+        Returns
+        -------
+        sorting_result: SortingResult
+            The SortingResult object
+
+        Examples
+        --------
+
+        >>> extension = sortres.compute("waveforms", **some_params)
+        >>> wfs = extension.data["waveforms"]
+
+        """
+        
+
+        extension_class = get_extension_class(extension_name)
+
+        # check dependencies
+        if extension_class.need_recording:
+            assert self.has_recording(), f"Extension {extension_name} need the recording"
+        for dependency_name in extension_class.depend_on:
+            ext = self.get_extension(dependency_name)
+            assert ext is not None, f"Extension {extension_name} need {dependency_name} to be computed first"
+        
+        extension_instance = extension_class(self)
+        extension_instance.set_params(save=save, **params)
+        extension_instance.run(save=save)
+        
+        self.extensions[extension_name] = extension_instance
+
+        # TODO : need discussion
+        return extension_instance
+        # OR
+        return extension_instance.data
+
+    def get_saved_extension_names(self):
+        """
+        Get extension saved in folder or zarr that can be loaded.
+        """
+        assert self.format != "memory"
+        global _possible_extensions
+
+        if self.format == "zarr":
+            zarr_root = self._get_zarr_root(mode="r")
+            if "extensions" in zarr_root.keys():
+                extension_group = zarr_root["extensions"]
+            else:
+                extension_group = None
+
+        saved_extension_names = []
+        for extension_class in _possible_extensions:
+            extension_name = extension_class.extension_name
+            if self.format == "binary_folder":
+                is_saved = (self.folder / extension_name).is_dir() and (self.folder / extension_name / "params.json").is_file()
+            elif self.format == "zarr":
+                if extension_group is not None:
+                    is_saved = extension_name in extension_group.keys() and "params" in extension_group[extension_name].attrs.keys()
+                else:
+                    is_saved = False
+            if is_saved:
+                saved_extension_names.append(extension_class.extension_name)
+
+        return saved_extension_names
+
+    def get_extension(self, extension_name: str):
+        """
+        Get a ResultExtension.
+        If not loaded then load is automatic.
+
+        Return None if the extension is not computed yet (this avoid the use of has_extension() and then get it)
```

Comment:
```suggestion
        Return None if the extension is not computed yet (this avoids the use of has_extension() and then get it)
```

not sure what then get it means.